objects/src/accounts/storage/mod.rs

```diff
@@ -271,6 +300,26 @@ impl Serializable for AccountStorage {
             target.write_u8(idx as u8);
             target.write(value);
         }
+
+        // serialize the number of StorageMaps
+        target.write_u8(self.maps.len() as u8);
+
+        // serialize storage maps
+        for storage_map in &self.maps {
+            let filled_leaves = storage_map
+                .entries()
+                .filter(|(_, value)| *value != SimpleSmt::<STORAGE_TREE_DEPTH>::EMPTY_VALUE)
+                .collect::<Vec<_>>();
+
+            // Write the number of filled leaves for this map
+            target.write_u8(filled_leaves.len() as u8);
+
+            // Write each (key, value) pair
+            for (key, value) in filled_leaves {
+                target.write(key);
+                target.write(value);
+            }
+        }
```

Comment:
**Request**s:

- Implement `Serializable` and `Deserializable` for the `Smt` in the `crypto` repo
- Implement `Serializable` and `Deserializable` for `StorageMap`
- Use the existing blanket implementation for `Vec<_>` here

Reply:
OK, so I have added it now for `StorageMap`. It would be cleaner to add `Serializable` and `Deserializable` to `Smt` in `miden-crypto`. 

@plafer Do you have an opinion? Do you want to serialize and deserialize the `Smt` in crypto also by only taking the `entries` as a vector an rebuilding the `Smt` at deserialization?